### PR TITLE
Iteration 1: User story 5

### PR DESCRIPTION
**Pull request by dhalverson**
**Adds**

- User model tests
- Updates nav bar to include admin view logic
- Adds admin routes
- Creates admin dashboard and user HTML placeholders
- Adds enum roles to user
- Creates admin controller

### CODE REVIEW

File: spec/features/navigation_spec.rb [hunk 1]
```diff
@@ -43,4 +43,36 @@
       end
     end
   end
+  describe 'As an Admin' do
```
Comment (dhalverson) [inline spec/features/navigation_spec.rb:46]: Need to add other types of users to this test for edge cases

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -12,4 +12,49 @@
     it { should validate_presence_of :password }
   end
 
+  describe "roles" do
```
Comment (dhalverson) [inline spec/models/user_spec.rb:15]: Need to test methods here as well

Comment (dhalverson) [inline spec/models/user_spec.rb:15]: Need to test the methods here

File: spec/features/navigation_spec.rb (hunk 1 above)
Comment (dhalverson) [inline spec/features/navigation_spec.rb:46]: Need to test edge case here, to test other types of users

**Review by dhalverson — COMMENTED** [review]
Need to update feature and model tests for this PR

File: app/controllers/admins_controller.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class AdminsController < ApplicationController
```
Comment (PhilipDeFraties) [inline app/controllers/admins_controller.rb:1]: Admin::DashboardController < Admin::BaseController I think will be the first line

**Review by PhilipDeFraties — CHANGES_REQUESTED** [review]
Looks good, I think you did the migration correctly. Just gotta look at the admin controllers and figure out the base/dashboard stuff.

File: config/routes.rb [hunk 4]
```diff
@@ -40,4 +40,7 @@
   get "/users/new", to: "users#new"
   post "/users/new", to: "users#create"
   get "/users/:id", to: "users#show"
+
+  get '/admin', to: 'admins#dashboard'
```
Comment (PhilipDeFraties) [inline config/routes.rb:44]: I think the routes here are a bit off, I dont think dashboard is supposed to be an action, I believe the route would look like this: 
`get '/admin/dashboard', to: "dashboard#index"`
This will then look for a controller called Admin::DashboardController

**Comment by dhalverson** [comment]
Awaiting US 8 PR to add this content
